Doc/c-api/long.rst

```diff
@@ -85,13 +85,12 @@ All integers are implemented as "long" integer objects of arbitrary size.
    Return a new :c:type:`PyLongObject` based on the string value in *str*, which
    is interpreted according to the radix in *base*.  If *pend* is non-*NULL*,
    *\*pend* will point to the first character in *str* which follows the
-   representation of the number.  If *base* is ``0``, the radix will be
-   determined based on the leading characters of *str*: if *str* starts with
-   ``'0x'`` or ``'0X'``, radix 16 will be used; if *str* starts with ``'0o'`` or
-   ``'0O'``, radix 8 will be used; if *str* starts with ``'0b'`` or ``'0B'``,
-   radix 2 will be used; otherwise radix 10 will be used.  If *base* is not
-   ``0``, it must be between ``2`` and ``36``, inclusive.  Leading spaces are
-   ignored.  If there are no digits, :exc:`ValueError` will be raised.
+   representation of the number.  If *base* is ``0``, *str* is interpreted using
+   the :ref:`integers` definition; a literal 0 evaluates to 0; since
+   leading zeroes are not allowed in a :token:`decinteger`, this raises a :exc:`ValueError`.
```

Comment:
The second and third parts about literal 0 and leading zeros still seem awkward. What about

If `*base*` is 0, `*str*` is interpreted using the :ref:`` `integers` `` definition. In this case, leading zero digits followed by a non-zero decimal digit raises a :exc:`` `ValueError` ``. If `*base*` is not 0, . . .